google/cloud/bigtable/client.py

```diff
@@ -95,7 +306,37 @@ def __init__(
         table_id: str,
         app_profile_id: str | None = None,
     ):
-        raise NotImplementedError
+        """
+        Initialize a Table instance
+
+        Must be created within an async run loop context
+
+        Args:
+            instance_id: The Bigtable instance ID to associate with this client
+                instance_id is combined with the client's project to fully
+                specify the instance
+            table_id: The ID of the table.
+            app_profile_id: (Optional) The app profile to associate with requests.
+                https://cloud.google.com/bigtable/docs/app-profiles
+        Raises:
+          - RuntimeError if called outside of an async run loop context
+        """
+        self.client = client
+        self.instance = instance_id
+        self.table_id = table_id
+        self.app_profile_id = app_profile_id
+        # raises RuntimeError if called outside of an async run loop context
+        try:
+            self._register_instance_task = asyncio.create_task(
+                self.client.register_instance(instance_id)
+            )
+        except RuntimeError:
+            warnings.warn(
+                "Table should be created in an asyncio event loop."
+                " Instance will not be registered with client for refresh",
+                RuntimeWarning,
+                stacklevel=2,
+            )
```

Comment:
should this error instead of just warn? Being able to initialize a Client outside an async context could make sense but should we allow going as far as initializing a Table without being in an async context?

Reply:
Is there are main use cases that have benefits where a user would want to initialize these outside of an async context and then manually register? Seems much more difficult and more work for the user.

Reply:
Yeah, I've been a little unsure about this too, but I think making the Table require an async context is probably a good middle ground. And that will allow us to change the registration methods into private methods

I'll make that change, thanks for the input